controllers/gate_controller_test.go

```diff
@@ -44,3 +47,108 @@ func TestHandleGetGateById(t *testing.T) {
 
 	assert.Equal(t, http.StatusOK, ctx.Writer.Status())
 }
+
+func TestHandleCreateNewGate(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockService := mocks.NewMockIGateRepository(mockCtrl)
+	controllerRepo := NewGateRepository(mockService)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	gate := models.Gate{
+		GateNumber:  1,
+		FloorNumber: 1,
+	}
+	mockService.EXPECT().CreateNewGate(&gate).Return(nil)
+	reqBody := `{"gate_number" : 1,"floor_number" : 1}`
+	ctx.Request, _ = http.NewRequest("POST", "/gate", strings.NewReader(reqBody))
+	controllerRepo.HandleCreateNewGate(ctx)
+
+	assert.Equal(t, http.StatusOK, ctx.Writer.Status())
+	var response models.Gate
+	err := json.Unmarshal([]byte(reqBody), &response)
+
+	assert.NoError(t, err)
+	assert.Equal(t, gate.GateNumber, response.GateNumber)
```

Comment:
1. shouldn't you be asserting this with the request payload and not the response payload?
2. in REST, CREATE call does not usually send the created data in the response body - only the id in a `location-id` header.

Reply:
- Request payload is in reqBody. 
- `err := json.Unmarshal([]byte(reqBody), &response)` unmarshals the JSON request body back into a gate object(response) to verify that the data sent in the request matches what was expected.
- I'll rename response to gateModel